src/printer.js

```diff
@@ -2967,12 +3009,28 @@ function printArgumentsList(path, options, print) {
     ]);
   }
 
+  const printedArgumentList = concat(
+    printed.reduce((result, printedArg, index) => {
```

Comment:
Please don't use reduce, this is super inefficient to build a new nested array for every single element.

I would get inspiration from how we print arrays: https://github.com/prettier/prettier/blob/master/src/printer.js#L4718